Source/JavaScriptCore/wasm/WasmAirIRGenerator.cpp

```diff
@@ -3562,10 +3565,27 @@ auto AirIRGenerator::addArrayGet(uint32_t typeIndex, ExpressionType arrayref, Ex
     // https://bugs.webkit.org/show_bug.cgi?id=245405
     emitCCall(&operationWasmArrayGet, getValue, instanceValue(), addConstant(Types::I32, typeIndex), arrayref, index);
 
-    switch (elementType.kind) {
+    switch (resultType.kind) {
     case TypeKind::I32:
         result = g32();
         append(Move32, getValue, result);
+        switch (arrayGetKind) {
+        case GCOpType::ArrayGet:
+        case GCOpType::ArrayGetU:
+            break;
+        case GCOpType::ArrayGetS: {
```

Comment:
Similar to the B3 changes, the signed get post-processing is emitted in-line here now.